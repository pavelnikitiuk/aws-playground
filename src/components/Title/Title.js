import { Box, Flex } from '@chakra-ui/react';
import React from 'react';
import RichText from '../RichText';
import { textPropTypes } from '../RichText/RichText';

function Title({ text, description, aside }) {
  return (
    <>
      <header>
        <Flex justifyContent="space-between" wrap="wrap">
          <Box flex="1" mb="10">
            <RichText text={text} />
            <RichText text={description} />
          </Box>
          <Box as="aside">
            <RichText text={aside} />
          </Box>
        </Flex>
      </header>
      <section />

    </>
  );
}

Title.propTypes = {
  text: textPropTypes.isRequired,
  description: textPropTypes.isRequired,
  aside: textPropTypes.isRequired,
};

export default Title;
