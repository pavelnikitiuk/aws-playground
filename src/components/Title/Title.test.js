import React from 'react';
import renderer from 'react-test-renderer';
import Title from './index';

const MOCK_TEXT = [];

describe('Given the Title component', () => {
  describe('when it is rendered', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = renderer.create(
        <Title text={MOCK_TEXT} aside={MOCK_TEXT} description={MOCK_TEXT} />,
      );
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
