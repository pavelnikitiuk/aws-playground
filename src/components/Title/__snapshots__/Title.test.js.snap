// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Title component when it is rendered should match the snapshot 1`] = `
Array [
  <header>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-vxgrp0{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;}",
        }
      }
      data-emotion="css vxgrp0"
    />
    <div
      className="css-vxgrp0"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-kzjg2w{-webkit-flex:1;-ms-flex:1;flex:1;margin-bottom:10px;}",
          }
        }
        data-emotion="css kzjg2w"
      />
      <div
        className="css-kzjg2w"
      />
      <aside
        className="css-0"
      />
    </div>
  </header>,
  <section />,
]
`;
